core/templates/pages/exploration-editor-page/translation-tab/translator-overview/translator-overview.component.spec.ts

```diff
@@ -58,6 +59,18 @@ describe('Translator Overview component', () => {
   let routerService: RouterService;
   let entityTranslationsService: EntityTranslationsService;
 
+  class MockUserExplorationPermissionsService {
+    static getPermissionsAsync() {
+      return Promise.resolve({
+        canVoiceover: true
+      });
+    }
+
+    fetchPermissionsAsync() {
+      return Promise.resolve();
+    }
+  }
```

Comment:
@Shivkant-Chauhan we can only spyOn or mock the service. We can't do both.

Reply:
I am facing errors while using the spyOn on the original function `UserExplorationPermissionsService`
error logs are : 
```
No overload matches this call.
  Overload 1 of 2, '(object: typeof UserExplorationPermissionsService, method: never): Spy<never>', gave the following error.
    Argument of type 'string' is not assignable to parameter of type 'never'.
```

should I let it as it is using the mock function - `MockUserExplorationPermissionsService`?

Reply:
@Shivkant-Chauhan Can you share the code also?

Reply:
```
    spyOn(UserExplorationPermissionsService, 'getPermissionsAsync').and
        .returnValue(Promise.resolve({
          canVoiceover: true
        }));
```
this error is on `getPermissionsAsync`
@gp201

Reply:
thanks @gp201 for telling! changed as suggested.